### PR TITLE
API-1513: Create metric about duplicated events

### PR DESCRIPTION
**Pull request by pierrejolly**
**Description (for Contributor and Core Developer)**

Add metric about duplicated events

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Component/Product/Webhook/ProductCreatedAndUpdatedEventDataBuilder.php [hunk 1]
```diff
@@ -71,6 +71,7 @@ public function build(BulkEventInterface $bulkEvent, UserInterface $user): Event
 
             $data = [
                 'resource' => $this->connectorProductNormalizer->normalizeConnectorProduct($product),
+                'version_id' => sprintf('%s%s', $product->identifier(), $product->updatedDate()->getTimestamp()),
```
Comment (pierrejolly) [inline src/Akeneo/Pim/Enrichment/Component/Product/Webhook/ProductCreatedAndUpdatedEventDataBuilder.php:74]: To remove

File: src/Akeneo/Pim/Enrichment/Component/Product/Webhook/ProductCreatedAndUpdatedEventDataBuilder.php [hunk 2]
```diff
@@ -71,6 +71,7 @@ public function build(BulkEventInterface $bulkEvent, UserInterface $user): Event
...
             ];
             $collection->setEventData($event, $data);
```
Comment (pierrejolly) [inline src/Akeneo/Pim/Enrichment/Component/Product/Webhook/ProductCreatedAndUpdatedEventDataBuilder.php:76]:
```suggestion
            $collection->setEventData($event, $data);
            $collection->setEventVersion($event, sprintf('%s%s', $product->identifier(), $product->updatedDate()->getTimestamp()));
```

**Review by pyaillet — APPROVED** [review]
GTM

File: src/Akeneo/Pim/Enrichment/Component/Product/Webhook/ProductCreatedAndUpdatedEventDataBuilder.php [hunk 3]
```diff
@@ -72,7 +72,9 @@ public function build(BulkEventInterface $bulkEvent, UserInterface $user): Event
             $data = [
                 'resource' => $this->connectorProductNormalizer->normalizeConnectorProduct($product),
             ];
-            $collection->setEventData($event, $data);
+            $dataVersion = sprintf('%s_%s_%s', 'product', $product->identifier(), $product->updatedDate()->getTimestamp());
```
Comment (tomglvng) [inline src/Akeneo/Pim/Enrichment/Component/Product/Webhook/ProductCreatedAndUpdatedEventDataBuilder.php:75]:
```suggestion
            $dataVersion = sprintf('product_%s_%s', $product->identifier(), $product->updatedDate()->getTimestamp());
```

File: src/Akeneo/Pim/Enrichment/Component/Product/Webhook/ProductModelCreatedAndUpdatedEventDataBuilder.php [hunk 4]
```diff
@@ -72,7 +72,9 @@ public function build(BulkEventInterface $bulkEvent, UserInterface $user): Event
             $data = [
                 'resource' => $this->connectorProductModelNormalizer->normalizeConnectorProductModel($productModel),
             ];
-            $collection->setEventData($event, $data);
+            $dataVersion = sprintf('%s_%s_%s', 'product_model', $productModel->code(), $productModel->updatedDate()->getTimestamp());
```
Comment (tomglvng) [inline src/Akeneo/Pim/Enrichment/Component/Product/Webhook/ProductModelCreatedAndUpdatedEventDataBuilder.php:75]:
```suggestion
            $dataVersion = sprintf('product_model_%s_%s', $productModel->code(), $productModel->updatedDate()->getTimestamp());
```

File: src/Akeneo/Pim/Enrichment/Component/Product/Webhook/ProductModelRemovedEventDataBuilder.php [hunk 5]
```diff
@@ -42,7 +42,9 @@ public function build(BulkEventInterface $bulkEvent, UserInterface $user): Event
                     'code' => $event->getCode()
                 ],
             ];
-            $collection->setEventData($event, $data);
+            $dataVersion = sprintf('%s_%s', 'product_model', $event->getCode());
```
Comment (tomglvng) [inline src/Akeneo/Pim/Enrichment/Component/Product/Webhook/ProductModelRemovedEventDataBuilder.php:45]:
```suggestion
            $dataVersion = sprintf('product_model_%s', $event->getCode());
```

File: src/Akeneo/Pim/Enrichment/Component/Product/Webhook/ProductRemovedEventDataBuilder.php [hunk 6]
```diff
@@ -42,7 +42,9 @@ public function build(BulkEventInterface $bulkEvent, UserInterface $user): Event
                     'identifier' => $event->getIdentifier()
                 ],
             ];
-            $collection->setEventData($event, $data);
+            $dataVersion = sprintf('%s_%s', 'product', $event->getIdentifier());
```
Comment (tomglvng) [inline src/Akeneo/Pim/Enrichment/Component/Product/Webhook/ProductRemovedEventDataBuilder.php:45]:
```suggestion
            $dataVersion = sprintf('product_%s', $event->getIdentifier());
```

**Review by tomglvng — APPROVED** [review]

**Pull request merged**